sjsonnet/src/sjsonnet/Std.scala

```diff
@@ -927,20 +927,25 @@ class Std(private val additionalNativeFunctions: Map[String, Val.Func] = Map.emp
           var i = 0
           while (i < keys.length) {
             val key = keys(i)
-            val lValue = if (l.containsVisibleKey(key)) l.valueRaw(key, l, pos)(ev) else null
             val rValue = if (r.containsVisibleKey(key)) r.valueRaw(key, r, pos)(ev) else null
             if (!rValue.isInstanceOf[Val.Null]) { // if we are not removing the key
-              if (lValue != null && rValue == null) {
-                // Preserve the LHS/target value:
-                kvs(kvsIdx) = (key, new Val.Obj.ConstMember(false, Visibility.Normal, lValue))
-              } else if (lValue.isInstanceOf[Val.Obj] && rValue.isInstanceOf[Val.Obj]) {
-                // Recursively merge objects:
-                kvs(kvsIdx) = (key, createLazyMember(recPair(lValue, rValue)))
-              } else if (rValue != null) {
+              if (l.containsVisibleKey(key)) {
+                if (rValue == null) {
+                  // Preserve the LHS/target value:
+                  kvs(kvsIdx) = (key, createLazyMember(l.valueRaw(key, l, pos)(ev)))
+                } else {
+                  lazy val lValue = l.valueRaw(key, l, pos)(ev)
```

Comment:
```suggestion
                  val lValue = l.valueRaw(key, l, pos)(ev)
```

Reply:
This is intentionally `lazy` in order to skip evaluation in when `rValue` is not an object, avoiding duplication between the "rValue is not an object` and "rValue is an object but lValue is not an object" cases.

Reply:
Thanks, better with an inline comments, I was thinking that the createLazyMember is already some kind of lazy.